persistence/resource.go

```diff
@@ -1,19 +1,42 @@
 package persistence
 
 import (
-	"github.com/go-openapi/strfmt"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
 )
 
 // Resource represents the resource as it exists in the persistence layer
 // this is very similar to models.Resource, but COULD vary, so we should
 // keep them separated
 type Resource struct {
-	ID        string     `json:"id"`
-	AtType    strfmt.URI `json:"@type"`
-	AtContext strfmt.URI `json:"@context"`
-	Access    string     `json:"access"`
-	Label     string     `json:"label"`
-	Preserve  bool       `json:"sdrPreserve"`
-	Publish   bool       `json:"publish"`
-	SourceID  string     `json:"source_id"`
+	data map[string]*dynamodb.AttributeValue
+}
+
+// NewResource returns a new instance of the resource
+func NewResource() *Resource {
```

Comment:
i do like this package / grouping of functions a bit better now. is there any reason to keep this separate from the serializer package? they seem to be attempting to manage basically our codebase's internal serialization of the object for the sake of code handling. Personally, I'd like to see that grouped as clearly as possible for sake of readability / comprehensibility.

Reply:
@cmh2166 Yeah, I'm happy to move this to the sterilizers package. That's a good idea.